### DISCUSSION TITLE
Use hidden and visible methods on SelectColumn

### DISCUSSION THREAD

**Discussion by myregistration** [filamentphp/filament#13423] [Help]
### Package

Table builder

### Package Version

v3

### How can we help you?

I want to hide/show SelectColumn based on a $record property, but the $record is null.  This seems impossible since it's a table of records.  I want to hide the SelectColumn and show a TextColumn (and vice versa) based on the user's ownership of the record.  This seems reasonable to me.

**Comment by micraux** [comment]
Hello,

If I understand well, you want the owner of the record to be the only one allowed to modify the value (SelectColumn).

If so, a simple solution would be to use `->disabled()`

**Reply by myregistration** [reply]
Thank you for your response. I actually don't want it to be displayed at all because the value set by another user might be something I don't want this user to see. I just want to display as text if it's not the auth user's record.

**Comment by dissto** [comment]
Perhaps you could do something like: 

```php
public static function table(Table $table): Table
{
    $condition = false; // whatever 
    $myCoolColumn = match ($condition) {
        false => SelectColumn::make('name'),
        default => TextColumn::make('name'),
    };
    return $table
        ->columns([
           $myCoolColumn,
        ])
```
🤔 

Or do you actually mean on a per row basis?

**Reply by myregistration** [reply]
The problem is I need access to the $record in order to set the condition

**Reply by dissto** [reply]
A lol my bad, missed the `$record` part 🤓

I don't think you can swap out the column at runtime on a per row basis.

Perhaps with a custom view where you would reuse the components? Not sure, haven't tested that. 🤔

**Reply by myregistration** [reply]
I created a custom table column by extending `SelectColumn` and added trait `CanBeHidden`, but still when access `$record` in the `hidden()` closure `dd($record)` is still null.

**Comment by micraux** [comment]
So, instead of `->disabled`, you could use `->hidden()` with a condition.

Could you share your code?
